src/Channels.cpp

```diff
@@ -29,8 +29,8 @@ int Channels::GetNumChannels()
       tinyxml2::XMLDocument doc;
       if (doc.Parse(response.c_str()) == tinyxml2::XML_SUCCESS)
       {
-        tinyxml2::XMLElement* channelsNode = doc.RootElement()->FirstChildElement("channels");
-        tinyxml2::XMLElement* pChannelNode;
+        tinyxml2::XMLNode* channelsNode = doc.RootElement()->FirstChildElement("channels");
```

Comment:
After looking at the tinyxml2 docs I'm not sure I understand switching to the use of XMLNode. A Node is simply a base class for many other types. What is the benefit of this vs using XMLElement?

FYI, everything that was read into a type of XMLElement is in fact an XMLElement so I'm not sure I see the benefit here.

Reply:
> 
> 
> FYI, everything that was read into a type of XMLElement is in fact an XMLElement so I'm not sure I see the benefit here.

It was simply because all the variable names say  "Node"  I felt it might be confusing to others who aren't used as I am that might want to use this XMLUtils.h.  Also the original Doxygen referenced nodes not elements so I wanted to be consistent with the documentation I readded.

Reply:
Ok, regardless of what it is named in the code everywhere it is used should be an XMLElement and not an XMLNode. So I would change the variable names and not the type of you wish.

Reply:
> 
> 
> Ok, regardless of what it is named in the code everywhere it is used should be an XMLElement and not an XMLNode. So I would change the variable names and not the type of you wish.

Why isn't it better to keep it the same as core, https://github.com/xbmc/xbmc/blob/master/xbmc/utils/XMLUtils.h and the old library in kodi-platform?

Reply:
Sorry, I thought your version was the same as core. I didn’t realise you had changed it.

I understand now.